liquidity/liquidity.go

```diff
@@ -1578,3 +1662,27 @@ func channelIsCustom(channel lndclient.ChannelInfo) bool {
 	// don't want to consider it for swaps.
 	return channel.CustomChannelData != nil
 }
+
+// getCustomAssetData returns the asset data for a custom channel.
+func getCustomAssetData(channel lndclient.ChannelInfo, assetId string,
+) *rfqmsg.JsonAssetChanInfo {
+
+	if channel.CustomChannelData == nil {
+		return nil
+	}
+
+	var assetData rfqmsg.JsonAssetChannel
+	err := json.Unmarshal(channel.CustomChannelData, &assetData)
+	if err != nil {
+		log.Errorf("Error unmarshalling custom channel data: %v", err)
+		return nil
```

Comment:
I think we should return the error here.

Reply:
Could you add the channel id to the log?